exp/api/v1alpha3/webhook_test.go

```diff
@@ -20,17 +20,23 @@ import (
 	"fmt"
 	"testing"
 
-	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/component-base/featuregate/testing"
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/feature"
 	"sigs.k8s.io/cluster-api/util"
+
+	. "github.com/onsi/gomega"
 )
 
 func TestMachinePoolConversion(t *testing.T) {
+	// NOTE: MachinePool feature flag is disabled by default, thus preventing to create or update MachinePool.
```

Comment:
Note: I think it's fine to add this as part of this PR, but we should open an issue to update how the conversion tests/webhook in the MachinePool is done. I'm not sure what our testing area is exactly if we have this test in v1alpha3, but nothing in v1alpha4.